base/shell/explorer/traydeskbtn.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Show Desktop tray button header file
+ * COPYRIGHT:   Copyright 2006-2007 Thomas Weidenmueller <w3seek@reactos.org>
+ *              Copyright 2018-2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ *              Copyright 2023 Ethan Rodensky <splitwirez@gmail.com>
+ */
+
+#pragma once
+
+#include "precomp.h"
+#include <commoncontrols.h>
+#include <uxtheme.h>
+
+// This window class name is CONFIRMED on Win10 by WinHier.
+static const WCHAR szTrayShowDesktopButton[] = L"TrayShowDesktopButtonWClass";
+#define TSDB_CLICK (WM_USER + 100)
+
+// The 'Show Desktop' button at edge of taskbar
+class CTrayShowDesktopButton :
+    public CWindowImpl<CTrayShowDesktopButton, CWindow, CControlWinTraits>
+{
+    LONG m_nClickedTime;
+    HTHEME m_hTheme;
+    HTHEME m_hFallbackTheme;
+    MARGINS m_ContentMargins;
+    SIZE m_inset;
+    HICON m_icon;
+    SIZE m_szIcon;
+    BOOL m_highContrastMode;
+    BOOL m_drawWithDedicatedBackground;
+    BOOL m_bHovering;
+
+public:
+    BOOL m_bPressed;
+    BOOL IsHorizontal;
+
+    DECLARE_WND_CLASS_EX(szTrayShowDesktopButton, CS_HREDRAW | CS_VREDRAW, COLOR_3DFACE)
+
+    BEGIN_MSG_MAP(CTrayShowDesktopButton)
+        MESSAGE_HANDLER(WM_LBUTTONDOWN, OnLButtonDown)
+        MESSAGE_HANDLER(WM_NCLBUTTONDOWN, OnLButtonDown)
+        MESSAGE_HANDLER(WM_LBUTTONUP, OnLButtonUp)
+        MESSAGE_HANDLER(WM_NCLBUTTONUP, OnLButtonUp)
+        MESSAGE_HANDLER(WM_SETTINGCHANGE, OnSettingChanged)
+        MESSAGE_HANDLER(WM_THEMECHANGED, OnThemeChanged)
+        MESSAGE_HANDLER(WM_WINDOWPOSCHANGED, OnWindowPosChanged)
+        MESSAGE_HANDLER(WM_PAINT, OnPaint)
+        MESSAGE_HANDLER(WM_PRINTCLIENT, OnPrintClient)
+        MESSAGE_HANDLER(WM_TIMER, OnTimer)
+        MESSAGE_HANDLER(WM_MOUSEMOVE, OnMouseMove)
+        MESSAGE_HANDLER(WM_DESTROY, OnDestroy)
+        MESSAGE_HANDLER(TSDB_CLICK, OnClick)
+    END_MSG_MAP()
+
+    CTrayShowDesktopButton();
+    INT WidthOrHeight() const;
+    HRESULT DoCreate(HWND hwndParent);
+    VOID Click();
+    BOOL PtInButton(LPPOINT pt);
```

Comment:
Aren't some of these other functions "const" too? (in the same sense as the WidthOrHeight one above)

Reply:
...functions can be `const`??

Reply:
In C++ this means that the function doesn't change any of the members of the class.

Reply:
Huh...well TIL that's a thing you can do.

Seems that `PtInButton` can be `const`, but the other two can't (doesn't even compile), so...I'll change `PtInButton` to be `const` and leave the other two as-is.

Reply:
yes, it seems PtInButton is the only candidate for that.
Maybe the GetTaskbar functon in the private list could satisfy that as well.

Reply:
Well that function no longer exists, so...if it ever was, it isn't anymore.